service/controller/resource/configmapmigration/create.go

```diff
@@ -129,25 +120,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 				return microerror.Mask(err)
 			}
 		}
-
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out if chartconfig %#q has been deleted", chartSpec.ChartName))
-
-		_, err = getChartConfigByName(chartConfigs.Items, chartSpec.ChartName)
-		if IsNotFound(err) {
```

Comment:
The problem is I was already using this trigger. 🤦‍♂️ 

To delete the old configmaps to clean up. I can't see a way to do both steps in cluster-operator.

But we could add a resource to chart-operator that does this clean up. @oponder @tomahawk28 WDYT?

Reply:
I agreed, but rather than simply delete all old configmaps, I would like to keep it as a different name. (e.g. `deprecated-user-values`). 
And how we patch the chart-operator that we already released in TCs?

Reply:
The problem with renaming is if we don't automate the deletion I don't think it will happen. But we could rename it and then have an alert for us to clean up manually.

For chart-operator we don't have to delete the CMs. Better to use the normal release process. We create a new version of chart-operator and the logic is there when they upgrade.

Reply:
So we remove the deletion logic and then add replacement logic to chart-operator?

This is the best option I can see unless I'm missing something.

Reply:
> So we remove the deletion logic and then add replacement logic to chart-operator?

Let's do this way. I would take part in creating new resource in chart-operator.

Reply:
Aside from the code change, maybe we can fix this by introducing a process change.

So trying to raise the point again that things have changed location, and possibly making the upgrade that would trigger this migration something that has to be done with an SE and a checklist.

Like:

1. Do you still have user-values in the tenant cluster being deployed by automation?
   a. Fix that
2. Do you have access to the control plane? because after this that will be the only place where you can access the configuration.

Reply:
@oponder Sorry I missed this somehow. But I agree we need a process change here.

I mentioned that in the suggestions in the PM. https://github.com/giantswarm/giantswarm/issues/11180